src/components/language-selector/LanguageSelector.vue

```diff
@@ -0,0 +1,60 @@
+<template>
+  <div class="language-selector">
+    <span>{{ resultsLanguageLabel }}</span>
+    <span v-for="(language, index) in languages" :key="language">
+      <span @click="updateUrl(language.languageId)" class="language">{{ language.languageName }}</span>
+      <span v-if="index < languages.length - 2">, </span>
+      <span v-else-if="index == languages.length - 2">{{orLabel}}&nbsp;</span>
+    </span>
+  </div>
+</template>
+
+<script>
+import { mapState, mapGetters } from "vuex";
+import { parseSearchQueryString } from '../../QueryString';
+export default {
+  name: "language-selector",
+  props: ["languageUri", "urlPath", "resultsLanguageLabel", "orLabel"],
+  components: {},
+  created() {},
+  mounted() {
+    this.languages = this.convertDictionaryToObject(
+      JSON.parse(this.$root.$store.state.config.languages)
+    );
+   
+    this.indexName = this.$root.$store.state.config.indexName;
+  },
+  data() {
+    return {
+      languages: null,
+      indexName: null,
+      urlPath: this.urlPath,
+    };
+  },
+  methods: {
+    updateUrl: function (language) {
+        this.$root.$store.commit("updateLanguage", language);
+        var searchQuery = this.indexName != null && this.indexName !="" ? "?language=" + language : "?language="+ language +"&indexName="+this.indexName;
+        var searchParams = parseSearchQueryString(searchQuery);
```

Comment:
I don't think setting a parsed search query string is required. Are there any implications if the fetchResults action is dispatched only with the update language state?